server/game/cards/09.6-SD/KnowTheTerrain.js

```diff
@@ -0,0 +1,37 @@
+const DrawCard = require('../../drawcard.js');
+const { CardTypes, Players, Locations } = require('../../Constants');
+
+class KnowTheTerrain extends DrawCard {
+    setupCardAbilities() {
+        this.interrupt({
+            title: 'Switch the attacked province with a facedown province',
+            when: {
+                onConflictDeclaredBeforeReveal: (event, context) => event.conflict.conflictProvince.facedown && 
+                    context.player.isDefendingPlayer() &&
+                    event.conflict.conflictProvince.location !== Locations.StrongholdProvince
+            },
+            target: {
```

Comment:
if the card doesn't target, you need to use a selectCard action (or promptForSelect if you're using a handler)